### ISSUE TITLE
`className` with trailing space breaks output

### ISSUE THREAD

**Issue by cameronmcefee** [styled-components/jest-styled-components#53]
Given the following component:

```
import React, { Component } from 'react';
import styled from 'styled-components';

class Test extends Component {
  render() {
    return <StyledTest className="foo ">Hello world</StyledTest>;
  }
}

const StyledTest = styled.div`color: red;`;

export default Test;
```
and the following test:

```
import React from 'react';
import Test from './Test';
import { mount } from 'enzyme';
import 'jest-styled-components';
import toJson from 'enzyme-to-json';

test('renders correctly', () => {
  const test = mount(<Test />);
  expect(toJson(test)).toMatchSnapshot();
});
```

The following snapshot is generated

```
// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
c1cc10c1.c1cc10c1ec1cc10c1rc1cc10c1Wc1cc10c1Ec1cc10c1xc1cc10c1Gc1cc10c1 c1cc10c1{c1cc10c1
c1cc10c1 c1cc10c1 c1cc10c1cc1cc10c1oc1cc10c1lc1cc10c1oc1cc10c1rc1cc10c1:c1cc10c1 c1cc10c1rc1cc10c1ec1cc10c1dc1cc10c1;c1cc10c1
c1cc10c1}c1cc10c1
c1cc10c1
c1cc10c1<c1cc10c1Tc1cc10c1ec1cc10c1sc1cc10c1tc1cc10c1>c1cc10c1
c1cc10c1 c1cc10c1 c1cc10c1<c1cc10c1sc1cc10c1tc1cc10c1yc1cc10c1lc1cc10c1ec1cc10c1dc1cc10c1.c1cc10c1dc1cc10c1ic1cc10c1vc1cc10c1
c1cc10c1 c1cc10c1 c1cc10c1 c1cc10c1 c1cc10c1cc1cc10c1lc1cc10c1ac1cc10c1sc1cc10c1sc1cc10c1Nc1cc10c1ac1cc10c1mc1cc10c1ec1cc10c1=c1cc10c1"c1cc10c1"c1cc10c1
c1cc10c1 c1cc10c1 c1cc10c1>c1cc10c1
c1cc10c1 c1cc10c1 c1cc10c1 c1cc10c1 c1cc10c1<c1cc10c1dc1cc10c1ic1cc10c1vc1cc10c1
c1cc10c1 c1cc10c1 c1cc10c1 c1cc10c1 c1cc10c1 c1cc10c1 c1cc10c1cc1cc10c1lc1cc10c1ac1cc10c1sc1cc10c1sc1cc10c1Nc1cc10c1ac1cc10c1mc1cc10c1ec1cc10c1=c1cc10c1"c1cc10c1 c1cc10c1sc1cc10c1cc1cc10c1-c1cc10c1bc1cc10c1dc1cc10c1Vc1cc10c1ac1cc10c1Jc1cc10c1ac1cc10c1 c1cc10c1ec1cc10c1rc1cc10c1Wc1cc10c1Ec1cc10c1xc1cc10c1Gc1cc10c1"c1cc10c1
c1cc10c1 c1cc10c1 c1cc10c1 c1cc10c1 c1cc10c1>c1cc10c1
c1cc10c1 c1cc10c1 c1cc10c1 c1cc10c1 c1cc10c1 c1cc10c1 c1cc10c1Hc1cc10c1ec1cc10c1lc1cc10c1lc1cc10c1oc1cc10c1 c1cc10c1wc1cc10c1oc1cc10c1rc1cc10c1lc1cc10c1dc1cc10c1
c1cc10c1 c1cc10c1 c1cc10c1 c1cc10c1 c1cc10c1<c1cc10c1/c1cc10c1dc1cc10c1ic1cc10c1vc1cc10c1>c1cc10c1
c1cc10c1 c1cc10c1 c1cc10c1<c1cc10c1/c1cc10c1sc1cc10c1tc1cc10c1yc1cc10c1lc1cc10c1ec1cc10c1dc1cc10c1.c1cc10c1dc1cc10c1ic1cc10c1vc1cc10c1>c1cc10c1
c1cc10c1<c1cc10c1/c1cc10c1Tc1cc10c1ec1cc10c1sc1cc10c1tc1cc10c1>c1cc10c1
`;
```

Note the space in the `className` of the component. I discovered this in a component that has a conditionally added class. I can solve the don't-end-with-a-space problem a handful of ways on my end, but this feels like something that probably would be considered a bug.

In larger components, this causes Jest to crash due to the absurdly long strings it generates.

**Comment by cameronmcefee** [comment]
I imagine it has something to do with [this](https://github.com/styled-components/jest-styled-components/blob/master/src/styleSheetSerializer.js#L18), which splits on '\s'. That would give you an empty item at the end of the array if it ends with a space, and probably also chokes if multi-spaces are present too.

**Comment by MicheleBertoli** [comment]
Thank you very much @cameronmcefee.
Good catch, and useful information in the issue.
This is fixed in v4.2.1.
